src/components/app/app.jsx

```diff
@@ -1,30 +1,105 @@
-import React from "react";
+import React, {PureComponent} from "react";
 import PropTypes from "prop-types";
+import {BrowserRouter, Route, Switch} from "react-router-dom";
 
 import Main from "../main/main.jsx";
+import Offer from "../offer/offer.jsx";
 
-const App = (props) => {
-  const {offersQuantity, offers} = props;
+class App extends PureComponent {
+  constructor(props) {
+    super(props);
 
-  return (
-    <Main
-      offersQuantity={offersQuantity}
-      offers={offers}
-    />
-  );
-};
+    this.handleOfferCardMouseEnter = this.handleOfferCardMouseEnter.bind(this);
+    this.handleOfferCardTitleClick = this.handleOfferCardTitleClick.bind(this);
+    this._renderApp = this._renderApp.bind(this);
+
+    this.state = {
+      step: -1,
+      offerHovered: null,
+      offerActive: null,
+    };
+  }
+
+  handleOfferCardTitleClick(offerInfo) {
+    this.setState((prevState) => ({
+      offerActive: offerInfo,
+      step: prevState.step + 1,
+    }));
+  }
+
+  handleOfferCardMouseEnter(offerInfo) {
+    this.setState({
+      offerHovered: offerInfo,
+    });
+  }
+
+  _renderApp() {
+    const {offersQuantity, offers} = this.props;
+    const {step} = this.state;
+
+    if (step === -1) {
```

Comment:
```suggestion
    if (this.state.offerActive) {
```
а степ тогда можно будет убрать